tests/bluetooth/host/keys/testing_common_defs.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** Keys "test" IDs associated with a BT address */
+#define BT_ADDR_ID_0 (0x00)
+#define BT_ADDR_ID_1 (0x55)
+#define BT_ADDR_ID_2 (0xAA)
+#define BT_ADDR_ID_3 (0xA5)
+#define BT_ADDR_ID_4 (0x5A)
+#define BT_ADDR_ID_5 (0xAC)
+
+/** Bluetooth LE device "test" addresses */
+#define BT_ADDR_LE_1 ((bt_addr_le_t[]){{BT_ADDR_LE_PUBLIC, {{0x0A, 0x89, 0x67, 0x45, 0x23, 0x01}}}})
+#define BT_ADDR_LE_2 ((bt_addr_le_t[]){{BT_ADDR_LE_PUBLIC, {{0x0B, 0x89, 0x67, 0x45, 0x23, 0x01}}}})
+#define BT_ADDR_LE_3 ((bt_addr_le_t[]){{BT_ADDR_LE_PUBLIC, {{0x0C, 0x89, 0x67, 0x45, 0x23, 0x01}}}})
+#define BT_ADDR_LE_4 ((bt_addr_le_t[]){{BT_ADDR_LE_PUBLIC, {{0x0D, 0x89, 0x67, 0x45, 0x23, 0x01}}}})
+#define BT_ADDR_LE_5 ((bt_addr_le_t[]){{BT_ADDR_LE_PUBLIC, {{0x0E, 0x89, 0x67, 0x45, 0x23, 0x01}}}})
+
+/** Bluetooth LE device "test" random addresses */
+#define BT_RPA_ADDR_LE_1                                                                           \
```

Comment:
You may want to refer to BT core specification 7B, 1.3.2.2. the two MSBs of RPA and NRPA random addresses are fixed. If there is code that's handling that properly, you'll probably want to have both valid and invalid address values.

Reply:
There is only one check there to check if the address type is random and an RPA

`struct bt_keys *bt_keys_find_irk(uint8_t id, const bt_addr_le_t *addr)
{
	__ASSERT_NO_MSG(addr != NULL);

	BT_DBG("%s", bt_addr_le_str(addr));

	if (!bt_addr_le_is_rpa(addr)) {
		return NULL;
	}
`
The function checks if it is an RPA or not, and the behavior doesn't depend on being NRPA or any other value.
So, I was interested in just turning that check into false and for that I used a public address